### DISCUSSION TITLE
Use of "OTel native" or "native" referring to OpenTelemetry

### DISCUSSION THREAD

**Discussion by theletterf** [open-telemetry/opentelemetry.io#3808] [Terminology]
We've around 230+ hits for "native" in our docs. It usually refers to native instrumentation, schemas, and so on.

I'd like to suggest that we come up with a new term and edit the docs accordingly. The term could be:

- OTel-ready
- OTel compatible
- Or just "OTel" as an adjective when we talk about specs or schemas.

I think "[native](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/n/native-language)" might mean lots of things that are unrelated to technical aspects, so I'd rather go with "OTel-ready" or "OTel compatible", as boring as they might sound. That has the added benefit that one knows what the thing is native about.

+CC @svrnm

**Comment by svrnm** [comment]
cc @open-telemetry/docs-approvers 

I like this proposal, although I like to iterate over the exact terminology a little bit. What we want to express is "this library makes use of the OpenTelemetry API directly, there are no instrumentation libraries or any other workarounds needed". 

I also wonder if we need to make this _specific_ to libraries, because right now we also have apps that use OpenTelemetry "natively".

**Comment by svrnm** [comment]
A few additional suggestions:

* OpenTelemetry integrated
* OpenTelemetry enabled
* OpenTelemetry compliant (I would rather not use that one, because it implies some additional checks & "certifications")
* OpenTelemetry prepared
* OpenTelemetry included
* OpenTelemetry integrated
* OpenTelemetry built-in (see https://opentelemetry.io/community/mission/#telemetry-should-be-built-in)
* OpenTelemetry inside (...)

**Comment by chalin** [comment]
Btw, just a suggestion: we could host discussions under the [Discussions](https://github.com/open-telemetry/opentelemetry.io/discussions) section of this repo :). That would help keep discussion topics out of our **already-busy Issues list**.

**Comment by svrnm** [comment]
> Btw, just a suggestion: we could host discussions under the [Discussions](https://github.com/open-telemetry/opentelemetry.io/discussions) section of this repo :). That would help keep discussion topics out of our **already-busy Issues list**.

There was some guidance in the otel community repo on when to use discussions and when not, let me double check on that, but overall I agree that we might use that instead of additional issues

**Comment by theletterf** [comment]
TIL! I'll use Discussions from now on.

**Comment by svrnm** [comment]
via https://github.com/open-telemetry/community/blob/main/docs/how-to-configure-new-repository.md:

> A note on discussions: Discussions are disabled by default when creating a new repository. Maintainers may enable discussions, but in doing so take on the responsibility of moderating conversations in accordance with the [code of conduct](https://github.com/open-telemetry/community/blob/main/code-of-conduct.md). Maintainers who enable discussions are encouraged to configure the discussion categories to suit their needs, e.g. by removing categories which overlap with issues.

**Comment by cartermp** [comment]
I would vote for otel-integrated.

**Reply by svrnm** [reply]
yes, while writing down that list otel-integrated was also my favourite

**Reply by svrnm** [reply]
looks like we have so far a preference for "opentelemetry integrated" or "otel-integrated".

**Comment by svrnm** [comment]
@open-telemetry/docs-approvers please make suggestions what you prefer so we can get to a consensus :-)

**Comment by svrnm** [comment]
Note, that "native instrumentation" is embedded in the spec:

https://github.com/open-telemetry/opentelemetry-specification/blob/9f0310fda53973e707f27e09da986f5f153f6491/specification/upgrading.md#library-maintainers

> Developers who maintain OSS libraries are encouraged to use the OpenTelemetry API to add instrumentation directly into their library code. We refer to this approach as "native instrumentation."

**Comment by theletterf** [comment]
@svrnm Can we suggest a change upstream? Are we free to redefine or wrap that term in the docs?

**Comment by danielgblanco** [comment]
I personally like the term OTel-native rather than OTel-integrated for cases where libraries directly use the OTel API to produce telemetry.

Integrating with OTel can be done in many ways not related to usage of the API (e.g. collectors, shims, OTLP exporters, following semconv, etc). I think OTel-native would signal more clearly to users that a particularly library is "designed to run within the OTel ecosystem", in the same way that a component can run natively on a platform (i.e. without any type of compatibility layer).

**Reply by svrnm** [reply]
> in the same way that a component can run natively on a platform (i.e. without any type of compatibility layer).

I think this is what the [link to the microsoft style guide](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/n/native-language) that the @theletterf was calling out to be ambiguous. Since it lacks any reasoning I checked some other style guides, especially the one from Google which we use "transitively":

https://developers.google.com/style/word-list#native

> When referring to software products, try to use a more precise term—for example, use built-in to describe a feature that's part of a product. 

Although, the red hat documentation guidelines have 2 different instances listed, one that shouldn't be used (native interface) and one that should be used (natively compiled code):

https://redhat-documentation.github.io/supplementary-style-guide/#_n

**Reply by danielgblanco** [reply]
The Google example specifically calls out "cloud-native" as being "not necessarily clear"... which we use everywhere. That's a different topic though. And the RedHat style guide uses "Kubernetes native" or "native messaging protocol" elsewhere in the guide. Definitely not an easy answer to this! 😅 

With that being said, I also think that OTel-integrated is probably the clearest one. I guess one could apply the OTel-integrated term to many things, in the context of a library, that means it uses the OTel API to describe itself.

**Reply by svrnm** [reply]
How can we push this discussion forward? No matter how we decide it would be good to agree on a consistent terminology.

**Reply by theletterf** [reply]
Are we free to decide within the SIG Comms on this? If that's the case, I'd say we do that and socialize our decision.

**Reply by svrnm** [reply]
We could but I _would_ prefer some consensus across the project.

**Reply by danielgblanco** [reply]
Next maintainers call?

**Reply by theletterf** [reply]
I feel like I should be there. When's the next?

**Reply by svrnm** [reply]
Next monday might be skipped since there is otel community day, so the week after?

**Comment by svrnm** [comment]
"Native" is now part of the spec and we will stick with it
